src/components/Experience.test.ts

```diff
@@ -0,0 +1,170 @@
+import { BattleOutcome } from "battlemovr";
+import { expect } from "chai";
+import { Clock } from "lolex";
+import { IListMenuProgress, MenuGraphr } from "menugraphr";
+import { Children } from "react";
+import { FullScreenPokemon } from "../";
+import { stubBlankGame } from "../fakes.test";
+import { IBattleInfo, IBattleTeam, IPartialBattleOptions, IPokemon } from "./Battles";
+import { IPlayer } from "./Things";
+
+const skipBattle = (clock: Clock, fsp: FullScreenPokemon, player: IPlayer): void => {
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+    fsp.inputs.keyDownA(player);
+    clock.tick(2000);
+};
+
+describe("Experience", () => {
+    it("Ensures a pokemon levels up through the function call", (): void => {
+        // Arrange
+        const { fsp } = stubBlankGame();
+        const pokemon: IPokemon = fsp.equations.newPokemon({
+            level: 5,
+            title: "CHARMANDER".split(""),
+        });
+
+        // Act
+        fsp.experience.levelup(pokemon);
+
+        // Assert
+        expect(pokemon.level).to.be.equal(6);
+    });
+    it("Ensures a pokemon gains experience through the function call and doesn't level up", (): void => {
+        // Arrange
+        const { fsp } = stubBlankGame();
+        const pokemon: IPokemon = fsp.equations.newPokemon({
+            level: 5,
+            title: "CHARMANDER".split(""),
+        });
+
+        // Act
+        const result = fsp.experience.gainExperience(pokemon, 50);
+
+        // Assert
+        expect(pokemon.experience).to.be.equal(175);
+        expect(result).to.be.equal(false);
```

Comment:
These test two very separate things: 

1. Returning the correct result
2. Applying the experience

Separate into two different unit tests.

Reply:
Here and elsewhere